CMakeLists.txt

```
@@ -99,7 +99,11 @@ set(PORTBLAS_GENERATED_SRC ${CMAKE_CURRENT_BINARY_DIR}/generated_src)
 set(PORTBLAS_INCLUDE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>)
 set(PORTBLAS_COMMON_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/common/include)
-set(PORTBLAS_SRC ${CMAKE_CURRENT_SOURCE_DIR}/src)
+if(INSTALL_HEADER_ONLY)
+  set(PORTBLAS_SRC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src> $<INSTALL_INTERFACE:src>)
+else()
+  set(PORTBLAS_SRC ${CMAKE_CURRENT_SOURCE_DIR}/src)
+endif()
```

This looks a bit odd. I think it would work if you use `$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src> $<INSTALL_INTERFACE:src>` in this file below instead of `${PORTBLAS_SRC}`. I think only one of these 2 expressions is needed even.

---

The current situation is a little bit more tricky.
The reason why in oneMKL portblas configuration there is a function to reset include path [here](https://github.com/oneapi-src/oneMKL/blob/develop/src/blas/backends/portblas/CMakeLists.txt#L156-L159) is that using only `$<INSTALL_INTERFACE:src>`, as it was before, doesn't work. Moreover, there is a little mismatch also in installing the library in header only mode and using with `FetchContent`. The problem of using only `line 103` introduced in this PR is that `${PORTBLAS_SRC}` variable is used in `generate_blas_object()` and it fails if it uses `line 103` solution. I am not sure why but it seems it doesn't expand it correctly.

---

But are you sure that `PORTBLAS_SRC` is used in multiple places when we set `INSTALL_HEADER_ONLY`? It looks to me that this [line](https://github.com/codeplaysoftware/portBLAS/pull/512/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR218) is the only place.

Functions like `generate_blas_object` expect this variable to be only a single path while the `install` function below can accept multiple paths. In other words it looks to me that we should have 2 different variables and we're trying to merge them in one. Since we need to set `$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src> $<INSTALL_INTERFACE:src>` only once for the `install` function, we might as well don't set a variable for it. Am I missing something?

---

It was used only there, but I tried to use it also in this [line](https://github.com/codeplaysoftware/portBLAS/pull/512/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR149).

Trying to fix it, I ended up with the same conclusion as you. We need two different variables because they serve two different needs. 
I would change above lines in this way:
```suggestion
if(INSTALL_HEADER_ONLY)
  set(PORTBLAS_INSTALL_SRC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src> $<INSTALL_INTERFACE:src>)
endif()
```
Use this new variable for header only purpose and let the other for everything else.